conan/tools/cmake/cmakedeps/templates/target_data.py

```diff
@@ -169,6 +175,22 @@ def _get_dependency_filenames(self):
 
         return ret
 
+    def _get_dependencies_find_modes(self):
+        ret = {}
+        if self.conanfile.is_build_context:
+            return ret
+        deps = self.conanfile.dependencies.filter({"build": False, "visible": True, "direct": True})
+        for dep in deps.values():
+            dep_file_name = get_file_name(dep, self.find_module_mode)
```

Comment:
Hi @lasote! Sorry, have seen PR only now (when it has already merged :)).

Here should be not `self.find_module_mode`, but something which depends on `values[find_mode.lower()]` from 191 line. Like `values[find_mode.values()] == "MODULE"` (or so) (doesn't matter what will be here for "none" find_mode). This is important for dependencies with different `cmake_file_name` and `cmake_module_file_name`.